fuel-p2p/src/orchestrator.rs

```diff
@@ -20,58 +19,58 @@ use crate::{
     service::{FuelP2PEvent, FuelP2PService},
 };
 
-type ResponseFuture = Pin<Box<dyn Future<Output = Option<(OutboundResponse, RequestId)>>>>;
-
 pub struct NetworkOrchestrator {
-    p2p_service: FuelP2PService,
+    p2p_config: P2PConfig,
 
     /// receives messages from different Fuel components
     rx_request_event: Receiver<P2pRequestEvent>,
+    rx_outbound_responses: Receiver<Option<(OutboundResponse, RequestId)>>,
 
     // senders
     tx_consensus: Sender<ConsensusBroadcast>,
     tx_transaction: Sender<TransactionBroadcast>,
     tx_block: Sender<BlockBroadcast>,
+    tx_outbound_responses: Sender<Option<(OutboundResponse, RequestId)>>,
 
-    db: Arc<dyn RelayerDb>,
-
-    outbound_responses: FuturesUnordered<ResponseFuture>,
+    db: Arc<dyn P2pDb>,
 }
 
 impl NetworkOrchestrator {
-    pub async fn new(
-        local_keypair: Keypair,
+    pub fn new(
         p2p_config: P2PConfig,
         rx_request_event: Receiver<P2pRequestEvent>,
 
         tx_consensus: Sender<ConsensusBroadcast>,
         tx_transaction: Sender<TransactionBroadcast>,
         tx_block: Sender<BlockBroadcast>,
 
-        db: Arc<dyn RelayerDb>,
-    ) -> Result<Self, Box<dyn Error>> {
-        let p2p_service = FuelP2PService::new(local_keypair, p2p_config).await?;
+        db: Arc<dyn P2pDb>,
+    ) -> Self {
+        let (tx_outbound_responses, rx_outbound_responses) = tokio::sync::mpsc::channel(100);
 
-        Ok(Self {
-            p2p_service,
+        Self {
+            p2p_config,
             rx_request_event,
+            rx_outbound_responses,
             tx_block,
             tx_consensus,
             tx_transaction,
+            tx_outbound_responses,
             db,
-            outbound_responses: Default::default(),
-        })
+        }
     }
 
-    pub async fn run(&mut self) {
+    pub async fn run(mut self) -> anyhow::Result<Self> {
```

Comment:
Since this returns a Result, can we add test coverage for the success and failure cases?

Reply:
The only case where this would fail is if the p2p configuration was incorrect. As for the success cases, the p2p service has been initialized and tested in dozens of p2p tests already.